lang/lucene-japanese/src/main/java/org/carrot2/language/japanese/JapaneseLanguageComponents.java

```
@@ -22,12 +38,80 @@
 public class JapaneseLanguageComponents extends SingleLanguageComponentsProviderImpl {
   public static final String NAME = "Japanese";
 
+  private static UserDictionary userDict;
+
+  static {
+    try (Reader reader =
+        new InputStreamReader(
+            JapaneseLanguageComponents.class
+                .getClassLoader()
+                .getResourceAsStream("org/carrot2/language/japanese/japanese.userdict.utf8"),
+            StandardCharsets.UTF_8)) {
+      userDict = UserDictionary.open(reader);
+    } catch (IOException | NullPointerException e) {
+      // TODO
```

I'd throw a runtime exception, even in the patch.